src/testing/reduced_order_pod_adaptation.h

```diff
@@ -10,60 +10,60 @@ namespace PHiLiP {
 namespace Tests {
 
 /// Burgers Rewienski snapshot
-    template <int dim, int nstate>
-    class ReducedOrderPODAdaptation: public TestsBase
-    {
-    public:
-        /// Constructor.
-        ReducedOrderPODAdaptation(const Parameters::AllParameters *const parameters_input);
+template <int dim, int nstate>
+class ReducedOrderPODAdaptation: public TestsBase
+{
+public:
+    /// Constructor.
+    ReducedOrderPODAdaptation(const Parameters::AllParameters *const parameters_input);
 
-        /// Run test
-        int run_test () const override;
-    };
+    /// Run test
+    int run_test () const override;
+};
 
+///Functional to take the integral of the solution
+template <int dim, int nstate, typename real>
+class BurgersRewienskiFunctional : public Functional<dim, nstate, real>
+{
+public:
+    using FadType = Sacado::Fad::DFad<real>; ///< Sacado AD type for first derivatives.
+    using FadFadType = Sacado::Fad::DFad<FadType>; ///< Sacado AD type that allows 2nd derivatives.
+public:
+    /// Constructor
+    BurgersRewienskiFunctional(
+            std::shared_ptr<PHiLiP::DGBase<dim,real>> dg_input,
+            std::shared_ptr<PHiLiP::Physics::PhysicsBase<dim,nstate,FadFadType>> _physics_fad_fad,
+            const bool uses_solution_values = true,
+            const bool uses_solution_gradient = false)
+            : PHiLiP::Functional<dim,nstate,real>(dg_input,_physics_fad_fad,uses_solution_values,uses_solution_gradient)
+    {}
```

Comment:
Shouldn't there be a semi-colon here after `{}` ?

Reply:
No, there's no need. It's just an empty constructor.